### PR TITLE
fix: Input now support refs

### PR DESCRIPTION
**Pull request by tibuurcio**
# Summary
This PR adds support for using Refs in our Input component the same way Antd's native Input supports it. With this we are able to support implementations that need to control input focus.

[Implementation was based on the one from Antd](https://github.com/ant-design/ant-design/blob/7abae5e973e50143b2b1b52731faea0eb6b58121/components/input/index.ts)

## Testing Plan
- [X] Was this tested locally? If not, explain why.
- Run storybook and run tests

https://github.com/mParticle/aquarium/assets/5959178/7523b676-efcb-48d7-abec-6f86b632dfa3

## Reference Issue (For mParticle employees only.  Ignore if you are an outside contributor)
- Closes https://go.mparticle.com/work/UNI-225

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -2,5 +2,6 @@
   "semi": false,
   "singleQuote": true,
   "arrowParens": "avoid",
-  "printWidth": 120
+  "printWidth": 120,
+  "bracketSameLine": true
```
Comment (tibuurcio) [inline .prettierrc:6]: With our printWidth size it doesn't make much sense to keep the default for jsxBracketSameLine as well.

This is the change this code does:
image

Comment (tibuurcio) [reply .prettierrc:6]: This made some changes to files outside of the scope of this PR. I'm thinking on removing this and introducing it in a new PR 🤔 

Let me know what you all think.

Comment (jared-dickman) [reply .prettierrc:6]: yea best to have linting updates in a separate pr where possible

**Comment by tibuurcio** [comment]
I'm gonna cherry pick this PR cause the merging of next messed it up.

**Review by jared-dickman — APPROVED** [review]

File: src/components/data-entry/Input/index.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+}
+
+const Input = InternalInput as CompoundedComponent
+
+Input.Group = AntInput.Group
+Input.Password = AntInput.Password
+Input.Search = AntInput.Search
+Input.TextArea = AntInput.TextArea
+
+export { Input, type InputRef, type IInputProps }
```
Comment (jared-dickman) [inline src/components/data-entry/Input/index.ts:19]: curious why we need an index for Input only, cant this go in src/components/index?

File: src/components/navigation/GlobalNavigation/WorkspaceSelector/WorkspaceSelector.tsx [hunk 3]
```diff
@@ -17,10 +18,6 @@ import { useEffect } from 'react'
 import { useMemo } from 'react'
 import { debounce, hasImageAtSrc } from 'src/utils/utils'
 import { getInitials } from 'src/utils/utils'
-
-// TODO: Need to make our Input component comply with forwardRef to be able to import it from src/components
```
Comment (jared-dickman) [inline src/components/navigation/GlobalNavigation/WorkspaceSelector/WorkspaceSelector.tsx:21]: **praise** TODONE!

File: src/components/data-entry/Input/index.ts (hunk 2 above)
Comment (tibuurcio) [reply src/components/data-entry/Input/index.ts:19]: I'm testing these to see if we can use the already present index

File: src/components/data-entry/Input/Input.tsx [hunk 4]
```diff
@@ -2,20 +2,17 @@ import { Input as AntInput } from 'antd'
...
-      <AntInput {...props} />
+      <AntInput {...props} ref={ref} />
     </ConfigProvider>
   )
 }
 
-Input.Group = AntInput.Group
-Input.Password = AntInput.Password
-Input.Search = AntInput.Search
-Input.TextArea = AntInput.TextArea
```
Comment (tibuurcio) [inline src/components/data-entry/Input/Input.tsx:19]: Try to keep this in the Input file if we can.

**Comment by tibuurcio** [comment]
I messed up this PR somehow, gotta learn that git stuff properly 🤦🏼‍♂️ 
image
Gonna open a new one.

**Comment by jared-dickman** [comment]
@tibuurcio spoilers - even linus doesnt  "knows git properly" 😂

**Comment by tibuurcio** [comment]
#169 fixed

**Pull request closed without merging**